app/presenters/menu/settings_loader.rb

```diff
@@ -0,0 +1,36 @@
+module Menu
+  class SettingsLoader < YamlLoader
+    include Singleton
+
+    def self.load
+      instance.load_from_settings
+    end
+
+    def load_from_settings
+      begin
+        settings = ::Settings.ui.custom_menu
+        items = (settings || []).map { |i| create_custom_item(HashWithIndifferentAccess.new(i)) }
+      rescue => e
+        # if we encounter an error while loading the menus, we ignore the whole settings
+        $log.error("Error loading custom menu from settings: #{e}")
```

Comment:
Prefer using `_log` from the Vmdb::Logging mixin as it tacks on more location information.